### PR TITLE
Added scripts to validate and fix duplicated and empty @ids

### LINKED ISSUE #3735: Validate duplication of UUIDs
based on https://github.com/SatelliteQE/testimony/issues/121

https://github.com/SatelliteQE/betelgeuse/blob/master/betelgeuse.py#L631

https://github.com/SatelliteQE/betelgeuse/blob/master/betelgeuse.py#L592-L611

1) get a list of all uuids in the whole project
2) Check for duplicates
   a) if dup found exit(1)
   b) if no dup exit(0)

Put that to run on travisCI.

### LINKED ISSUE #3834: Duplicate test UUID across the test suite
I spotted some duplicate UUIDs while doing my routine check. I guess we should fix them to keep Polarion gods happy ;):
```bash
$ grep "@id" */*.py | sort | uniq -d
api/test_repository.py:        @id: 6dff0c90-170f-40b9-9347-8ec97d89f2fd
api/test_variables.py:        @id: a90b5bcd-f76c-4663-bf41-2f96e7e15c0f
cli/test_hostgroup.py:        @id: 32be4630-0032-4f5f-89d4-44f8d05fe585
cli/test_hostgroup.py:        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
ui/test_contenthost.py:        @id: 13b9422d-4b7a-4068-9a57-a94602cd6410
ui/test_hostcollection.py:        @id: 5a6fff0a-686f-419b-a773-4d03713e47e9
ui/test_remoteexecution.py:        @id: 7f0cdd1a-c87c-4324-ae9c-dbc30abad217
ui/test_repository.py:        @id: 8037a68b-66b8-4b42-a80b-fb08495f948d
```

### PR DESCRIPTION
**Pull request by rochacbruno**
closes #3834
closes #3735 

I fixed the duplications found by @rplevka  in #3834 

Added `make uuid-check` (based in @rplevka snippet and using @elyezer help)  which exits with status code `1` if any duplicated or empty `@id:` testimony tag is found in our code base. (added that script call to `make test-docstring` which is called by travis.

Added make commands to help contributor to check invalid `@id` before commiting.
(optionally can be added as a commit hook)

# Before commiting please do:

```
make test-docstring
make test-robottelo
make uuid-check
``` 

We can create a target to group all `make test-commit` what about?

# if duplicates found 

you will see the filename and duplicated `@id` in output so you can fix it.

If want to fix automatically use:

`make uuid-replace-duplicate` 

All duplicated (last-occurrence) will be replaced with new generated uuid)

# Writing new stubs

If writing new test-stubs you can leave the `@id:` empty as in

```
def test_something(self):
    """
    @id:
    @Assert: Test something 
    """
```

Then run

```make uuid-replace-empty```

A new uuid will be generated and placed in that empty tag.

Backup~ files will be created in same folder for each changed file.

This script uses the command `uuidgen` ensure you have it installed.

Not forget to mention that [similar idea](https://github.com/san7ket/uuid) has been implemented by @san7ket, so thanks for the script you emailed, it helped.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,6 @@
 
 # For Pycharm
 /.idea
+
+# For backup files.
+*~
```
Comment (rochacbruno) [inline .gitignore]: I added `~` here as our `uuid-replace-*` scripts created backup files and we do not want to push it to the repo.

**Comment by oshtaier** [comment]
Very nice. ACK

**Comment by sghai** [comment]
ACK

**Comment by sghai** [comment]
Let's see the magic of:
`make uuid-check`
`make uuid-replace-duplicate`
`make uuid-replace-empty`

**Pull request merged**